src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -599,24 +599,26 @@ private void WritePropertyEventTrace(int eventId, DependencyProperty property, D
 		[MethodImpl(MethodImplOptions.AggressiveInlining)]
 		private void ApplyPrecedenceOverride(ref DependencyPropertyValuePrecedences precedence)
 		{
-			if (_isSettingAProperty)
+			if(_precedenceOverride == DependencyPropertyValuePrecedences.ExplicitStyle)
```

Comment:
I don't think that there should be a special case for `ExplicitStyle`. Properties set from within property-changed callbacks should still be applied at local precedence, whatever type of style is triggering the callback.

Reply:
You should probably add a variation of the `When_Set_Within_Style_Application()` test where the callback is triggered by an explicit style.

Reply:
Nope... that's the problem: `<Setter>`s MUST apply their values using the `ExplicitStyle` precedence. That's the origin of this bug: the first setter were applying its value using the `Local` precedence (another bug were preventing other setters from applying the overriden precedence).

Explanation: in the previous version, while manually setting the `Style` property (which uses `Local` precedence), it was setting both `_precedenceOverride` to `ExplicitStyle` and the `_isSettingAProperty` to true, preventing the setters from applying there values under correct explicit precedence.

But I think there's a remaining bug: only the "first level" of setters should using the overriden precedence. I'll need to add a test fot it.

Reply:
I originally thought of animations, but yes it makes sense.

Reply:
TL;DR: you're right, except when this property is the `Style` itself.  In that case, the override for `ExpliciteStyle` precedence should get prioritize over the `Local` precedence.

Reply:
> there's a remaining bug: only the "first level" of setters should using the overriden precedence.

Well yeah, that's what I'm getting at.

I thought after we talked about this, that it would be easiest if `FrameworkElement.OnStyleChanged(Style oldStyle, Style newStyle, DependencyPropertyValuePrecedences precedence)` were to reset `DependencyObjectStore._isSettingAProperty` to false, which should handle the case of setting the `Style` property, but I guess you identified problems with that approach?

Reply:
Please add the following test:

```csharp
		[TestMethod]
		public void When_Set_Style_Property_PrecedenceInContext_No_Local()
		{
			var style = new Style();
			style.Setters.Add(new Setter(Border.BorderThicknessProperty, 10d));

			var sut = new Border { /*Don't set local BorderThickness*/ };

			using var registration1 = sut.RegisterDisposablePropertyChangedCallback(
				Border.BorderThicknessProperty,
				(dependencyObject, args) =>
				{
					sut.Tag = args.NewPrecedence;
				});

			using var registration2 = sut.RegisterDisposablePropertyChangedCallback(
				FrameworkElement.TagProperty,
				(dependencyObject, args) =>
				{
					args.NewPrecedence.Should().Be(DependencyPropertyValuePrecedences.Local);
					args.NewValue.Should().Be(DependencyPropertyValuePrecedences.ExplicitStyle);
				});

			using var _ = new AssertionScope();

			sut.Tag.Should().BeNull("After applying style");

			sut.Style = style;
		}
```

Reply:
@davidjohnoliver I just pushed a new version which should solve the problem you identified.  You test is also integrated in it.